Project1/script.js

```diff
@@ -1,110 +1,73 @@
-//Add new feature
-let isResultDisplayed = false; // Flag to track if a result has been displayed
-
-// Constants
-const PI = 3.1416;
-const E = 2.7182;
-
-// Variable storage
-let variables = {};
-
-// Update display function
-function updateDisplay(value) {
-    document.getElementById('display').value = value;
-}
-
-// Append value to display
-function appendToDisplay(value) {
-    let display = document.getElementById('display');
-
-    if (isResultDisplayed) {
-        // If a result has been displayed, reset display for new input
-        display.value = ''; 
-        isResultDisplayed = false;
+const constants = {
+    PI: Math.PI,
+    e: Math.E
+};
+
+const variables = {};
+
+// function appendToExpression(value) {
+//     const expressionInput = document.getElementById('expression');
+//     if (expressionInput.value === 'Error') {
+//         expressionInput.value = '';
+//     }
+//     expressionInput.value += value;
+// }
+
+function appendToExpression(value) {
+    const expressionInput = document.getElementById('expression');
+    if (expressionInput.disabled) 
+        //return; // Prevent input if disabled
+    expressionInput.disabled = false;
+     if (expressionInput.value === 'Error') {
+        expressionInput.value = '';
     }
-
-    display.value += value;
+    expressionInput.value += value;
 }
 
-// Clear display
-function clearDisplay() {
-    updateDisplay('');
-    isResultDisplayed = false; // Reset flag when display is cleared
+function deleteLast() {
+    const expressionInput = document.getElementById('expression');
+    expressionInput.value = expressionInput.value.slice(0, -1);
 }
 
-// Delete the last character from the display
-function deleteLastCharacter() {
-    let display = document.getElementById('display');
-    let currentValue = display.value;
-
-    if (currentValue.length > 0) {
-        // Remove the last character
-        display.value = currentValue.slice(0, -1);
-    }
+function clearExpression() {
+    document.getElementById('expression').value = '';
+    document.getElementById('output').textContent = '0.0000';
 }
 
-// Add variable
-function addVariable() {
-    let name = document.getElementById('varName').value.trim();
-    let value = parseFloat(document.getElementById('varValue').value.trim());
-
-    if (name === 'pi' || name === 'e') {
-        alert("Cannot use names 'pi' or 'e' for variables.");
-        return;
-    }
-
-    if (!isNaN(value)) {
-        variables[name] = value;
-        document.getElementById('varName').value = '';
-        document.getElementById('varValue').value = '';
-        alert(`Variable ${name} added with value ${value}`);
-    } else {
-        alert('Invalid value for variable.');
-    }
-}
+function calculateResult() {
+    const expressionInput = document.getElementById('expression');
+    let expression = expressionInput.value;
 
-// Evaluate the expression
-function evaluateExpression(expression) {
```

Comment:
The code quality looks good and I tested the application and here are some points that I think should be improved:

- Multiple operators should be allowed to be added one after the other for example: 2++2---
- Similarly user should not be able to input multiple decimal points together 

Other parts look good at first sight.